alsaaudio.pyi

```diff
@@ -0,0 +1,128 @@
+from typing import list
+
+PCM_PLAYBACK: int
+PCM_CAPTURE: int
+
+PCM_NORMAL: int
+PCM_NONBLOCK: int
+PCM_ASYNC: int
+
+PCM_FORMAT_S8: int
+PCM_FORMAT_U8: int
+PCM_FORMAT_S16_LE: int
+PCM_FORMAT_S16_BE: int
+PCM_FORMAT_U16_LE: int
+PCM_FORMAT_U16_BE: int
+PCM_FORMAT_S24_LE: int
+PCM_FORMAT_S24_BE: int
+PCM_FORMAT_U24_LE: int
+PCM_FORMAT_U24_BE: int
+PCM_FORMAT_S32_LE: int
+PCM_FORMAT_S32_BE: int
+PCM_FORMAT_U32_LE: int
+PCM_FORMAT_U32_BE: int
+PCM_FORMAT_FLOAT_LE: int
+PCM_FORMAT_FLOAT_BE: int
+PCM_FORMAT_FLOAT64_LE: int
+PCM_FORMAT_FLOAT64_BE: int
+PCM_FORMAT_MU_LAW: int
+PCM_FORMAT_A_LAW: int
+PCM_FORMAT_IMA_ADPCM: int
+PCM_FORMAT_MPEG: int
+PCM_FORMAT_GSM: int
+PCM_FORMAT_S24_3LE: int
+PCM_FORMAT_S24_3BE: int
+PCM_FORMAT_U24_3LE: int
+PCM_FORMAT_U24_3BE: int
+
+PCM_TSTAMP_NONE: int
+PCM_TSTAMP_ENABLE: int
+
+PCM_TSTAMP_TYPE_GETTIMEOFDAY: int
+PCM_TSTAMP_TYPE_MONOTONIC: int
+PCM_TSTAMP_TYPE_MONOTONIC_RAW: int
+
+PCM_FORMAT_DSD_U8: int
+PCM_FORMAT_DSD_U16_LE: int
+PCM_FORMAT_DSD_U32_LE: int
+PCM_FORMAT_DSD_U32_BE: int
+
+PCM_STATE_OPEN: int
+PCM_STATE_SETUP: int
+PCM_STATE_PREPARED: int
+PCM_STATE_RUNNING: int
+PCM_STATE_XRUN: int
+PCM_STATE_DRAINING: int
+PCM_STATE_PAUSED: int
+PCM_STATE_SUSPENDED: int
+PCM_STATE_DISCONNECTED: int
+
+MIXER_CHANNEL_ALL: int
+
+MIXER_SCHN_UNKNOWN: int
+MIXER_SCHN_FRONT_LEFT: int
+MIXER_SCHN_FRONT_RIGHT: int
+MIXER_SCHN_REAR_LEFT: int
+MIXER_SCHN_REAR_RIGHT: int
+MIXER_SCHN_FRONT_CENTER: int
+MIXER_SCHN_WOOFER: int
+MIXER_SCHN_SIDE_LEFT: int
+MIXER_SCHN_SIDE_RIGHT: int
+MIXER_SCHN_REAR_CENTER: int
+MIXER_SCHN_MONO: int
+
+VOLUME_UNITS_PERCENTAGE: int
+VOLUME_UNITS_RAW: int
+VOLUME_UNITS_DB: int
+
+def pcms(pcmtype:int) -> list[str]: ...
+def cards() -> list[str]: ...
+def mixers(cardindex:int=-1, device:str='default') -> list[str]: ...
+def asoundlib_version() -> str: ...
+
+class PCM:
```

Comment:
some functions are definitely still missing here.

Reply:
I didn't document the ones that return `None` and have no arguments, is that it? 

I think type hints are great, but as they're optional, less might be more. In any case, I'll try to double check for consistency

Reply:
i didn't check, i assume you're right.
but the argument doesn't convince me. firstly, that inconsistency is confusing - the question whether this was omitted by accident will come up each time someone notices it. secondly, i think that file may be used for auto-completion by some IDEs, so having complete declarations is helpful regardless.